### PR TITLE
feat: PredictableRandomizer will create random values based on a seed [TECH-1449]

### PR DESCRIPTION
**Pull request by enricocolasante**

### CODE REVIEW

**Review by radnov — APPROVED** [review]
Looks good to me. 👍  

Just curious about how the `xoroshiro128` implementations compares to Java's own Random (or SecureRandom) classes for this use case?

**Comment by enricocolasante** [comment]
> Looks good to me. +1
> 
> Just curious about how the `xoroshiro128` implementations compares to Java's own Random (or SecureRandom) classes for this use case?

I actually removed it for now. If needed we can think again of using it. I think it was better on performance than the standard Random classes from java.util

File: src/main/java/org/hisp/dhis/utils/Randomizer.java [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+     *
+     * @param list a List
+     * @param size the size of the sublist
+     * @param <T>  type
+     * @return a random sublist
+     */
+    <T> List<T> randomElementsFromList(List<T> list, int size);
+
+    /**
+     * Returns random string containing 6 alphabetical characters.
```
Comment (jbee) [inline src/main/java/org/hisp/dhis/utils/Randomizer.java:48]: does this mean the string is always 6 characters long or just that it has at least 6 letters?

File: src/main/java/org/hisp/dhis/utils/Randomizer.java [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+
+    /**
+     * Generates random national id
+     *
+     * @return a national id
+     */
+    String randomNationalId();
+
+    /**
+     * Generates random birthday date
```
Comment (jbee) [inline src/main/java/org/hisp/dhis/utils/Randomizer.java:124]: Any guarantees about the age? Maybe have a age range parameter?

**Pull request merged**